packages/core/src/browser/consoleObservable.ts

```diff
@@ -0,0 +1,96 @@
+import { callMonitored } from '../domain/internalMonitoring'
+import { computeStackTrace } from '../domain/tracekit'
+import { createHandlingStack, ErrorSource, formatErrorMessage, toStackTraceString } from '../tools/error'
+import { mergeObservables, Observable } from '../tools/observable'
+import type { ClocksState } from '../tools/timeUtils'
+import { clocksNow } from '../tools/timeUtils'
+import { find, jsonStringify } from '../tools/utils'
+
+const StatusType = {
+  debug: 'debug',
+  info: 'info',
+  warn: 'warn',
+  error: 'error',
+} as const
+
+type StatusType = typeof StatusType[keyof typeof StatusType]
+
+type ApiNameType = 'log' | 'debug' | 'info' | 'warn' | 'error'
+
+const LogStatusForApi = {
+  log: StatusType.info,
+  debug: StatusType.debug,
+  info: StatusType.info,
+  warn: StatusType.warn,
+  error: StatusType.error,
+}
+
+export interface ConsoleLog {
+  startClocks: ClocksState
+  message: string
+  status: StatusType
+  source: 'console'
+  stack?: string
+  handlingStack?: string
+}
+
+const consoleObservables: { [k in ApiNameType]?: Observable<ConsoleLog> } = {}
+
+export function initConsoleObservable(apis: ApiNameType[]) {
+  const observables = apis.map((api) => {
+    if (!consoleObservables[api]) {
+      consoleObservables[api] = createConsoleObservable(api)
+    }
+    return consoleObservables[api]!
+  })
+
+  return mergeObservables<ConsoleLog>(...observables)
+}
+
+/* eslint-disable no-console */
+function createConsoleObservable(api: ApiNameType) {
+  const observable = new Observable<ConsoleLog>(() => {
+    const originalConsoleApi = console[api]
+
+    console[api] = (...params: unknown[]) => {
+      const handlingStack = createHandlingStack()
+      callMonitored(() => {
+        originalConsoleApi.apply(console, params)
+        observable.notify(buildConsoleLog(params, api, handlingStack))
+      })
+    }
```

Comment:
> Won't be easy to use instrumentMethod because createHandlingStack is also used in [here](https://github.com/DataDog/browser-sdk/blob/462592a85a86cb2bfebd6b60808703add65b54a5/packages/rum-core/src/boot/rumPublicApi.ts#L193). It as to be at the same call stack depth everywhere it's called.

Yup! definitly not easy.

> 
> I moved the original console api call outside of callMonitored

Great!